minimal.js

```diff
@@ -83,36 +82,50 @@ module.exports = function (dirname, keys, opts) {
 
   var append = db.rawAppend = db.append
   db.post = Obv()
-  var queue = AsyncWrite(function (_, cb) {
-    var batch = state.queue
-    state.queue = []
-    append(batch, function (err, v) {
+
+  let writing = false
+
+  const write = () => {
+    writing = true
+    // Very defensive: Is this necessary? I don't know whether it's possible
+    // for another function to `state.queue.push()` between these two lines.
+    const batch = state.queue.slice()
+    state.queue = state.queue.slice(batch.length)
+
+    append(batch, function (err) {
+      // Previously this error wasn't being caught anywhere. :(
+      // New behavior: if a write fails, crash loudly and throw an error.
+      if (err) throw err
+
+      // If we have new messages in the queue, write them!
+      // Otherwise, run all callbacks added via `flush.add()`
+      if (state.queue.length) {
+        write()
+      } else {
+        writing = false
+        flush.runAll()
+      }
+
+      // Update the observable
       batch.forEach(function (data) {
         db.post.set(u.originalData(data))
       })
-      cb(err, v)
     })
-  }, function reduce (_, msg) {
-    return V.append(state, hmacKey, msg)
-  }, function (_state) {
-    return state.queue.length > 1000
-  }, function isEmpty (_state) {
-    return !state.queue.length
-  }, 100)
-
-  queue.onDrain = function onDrain () {
-    if (state.queue.length === 0) {
-      flush.runAll()
+  }
```

Comment:
:question: behaviour change - your queue no longer has a timer, as in it doesn't wait a minimum time between each write call. I don't know if that was needed / if there are any implications of removing.
I don't think it's dangerous though

Reply:
I think that came from Async-Write's goal to "keep a buffer ready to write at all times", whereas this code has the goal to **immediately write all queued messages to the database**. I've placed the queue limit and timers off to the side in my YAGNI bucket until we can write some tests or benchmarks that show why they're useful.

Reply:
Shall we leave an issue on the repo with notes about what was changed as an FYI for future spelunkers?